distributed/client.py

```diff
@@ -382,10 +382,11 @@ def __setstate__(self, state):
     def __del__(self):
         try:
             self.release()
-        except AttributeError:
+        except AttributeError as e:
             # Ocassionally we see this error when shutting down the client
             # https://github.com/dask/distributed/issues/4305
-            pass
+            if not shutting_down():
+                raise e
```

Comment:
I think you want to drop the `as e` here and just call `raise`

If done within an except block this raises the original exception with a clean traceback I think.

Reply:
thanks @mrocklin -- fixed in #0455fa8